File: algorithms/medium/two_city_scheduling.ex
defmodule Solution do
  @spec two_city_sched_cost(costs :: [[integer]]) :: integer
  def two_city_sched_cost(costs) do
    n = Enum.count(costs) |> div(2)

    costs
    |> Enum.sort(fn [a0, b0], [a1, b1] -> a0 - b0 < a1 - b1 end)
    |> Enum.with_index()
    |> Enum.reduce(0, fn {[a, b], i}, acc ->
      acc + if i < n, do: a, else: b
    end)
  end
end

File: algorithms/medium/two_city_scheduling.rs
impl Solution {
    pub fn two_city_sched_cost(mut costs: Vec<Vec<i32>>) -> i32 {
        costs.sort_by(|a, b| (a[0]-a[1]).cmp(&(b[0]-b[1])));
        let n = costs.len() / 2;
        let mut ans = 0;
        for i in 0..n {
            ans += costs[i][0];
            ans += costs[i+n][1];
        }
        ans
    }
}
